skbio/sequence/distance.py

```diff
@@ -113,3 +105,76 @@ def hamming(seq1, seq2):
         distance = scipy.spatial.distance.hamming(seq1.values, seq2.values)
 
     return float(distance)
+
+
+@experimental(as_of='0.4.2-dev')
+def kmer_distance(seq1, seq2, k, overlap=True):
+    """Compute the kmer distance between a pair of sequences
+
+    The kmer distance between two sequences is the fraction of kmers that are
+    unique to either sequence.
+
+    Parameters
+    ----------
+    seq1, seq2 : Sequence
+        Sequences to compute kmer distance between.
+    k : int
+        The kmer length.
+    overlap : bool, optional
+        Defines whether the kmers should be overlapping or not.
+
+    Returns
+    -------
+    float
+    kmer distance between seq1 and seq2.
+
+    Raises
+    ------
+    ValueError
+        If `k` is less than 1.
+    TypeError
+    If `seq1` and `seq2` are not ``Sequence`` instances.
+    TypeError
+    If `seq1` and `seq2` are not the same type.
+
+    Notes
+    -----
+    kmer counts are not incorporated in this distance metric.
+
+    ``np.nan`` will be returned if there are no kmers defined for the
+    sequences.
+
+    Examples
+    --------
+    >>> from skbio import Sequence
+    >>> seq1 = Sequence('ATCGGCGAT')
+    >>> seq2 = Sequence('GCAGATGTG')
+    >>> kmer_distance(seq1, seq2, 3)
+    0.9230769230769231
```

Comment:
Can you use the doctest ellipsis directive to prevent this doctest from failing if the number if slightly different on different systems? See `skbio.stats.composition.inner` for an example.